Neos.Flow/Classes/ObjectManagement/Proxy/Compiler.php

```diff
@@ -349,4 +341,46 @@ protected static function renderOptionArrayValueAsString(array $optionValue)
         }
         return '{ ' . implode(', ', $values) . ' }';
     }
+
+    /**
+     * Appends ORIGINAL_CLASSNAME_SUFFIX to the original class name
+     *
+     * @param string $classCode
+     * @param string $pathAndFilename
+     * @return string
+     * @throws Exception
+     */
+    protected function replaceClassName(string $classCode, string $pathAndFilename): string
+    {
+        $tokens = token_get_all($classCode);
+        $previousToken = [0 => null];
+        foreach ($tokens as $token) {
+            if (isset($classToken) && is_array($token) && $token[0] === T_STRING) {
+                $detectedClassName = $token[1];
+                $lines = explode(PHP_EOL, $classCode);
+
+                // We can't allow Unicode characters for class names even though PHP does, because such class names
+                // don't qualify as valid cache entry identifiers.
+                $classTokenLine = preg_replace('/([a-z\s]*?)(final\s+)?(interface|class)\s+([a-zA-Z0-9_]+)/', '$1$3 $4' . self::ORIGINAL_CLASSNAME_SUFFIX, $lines[$token[2] - 1]);
```

Comment:
It becomes obvious the test must break this, we identify the line with the class token correctly, but then do preg replace without taking the position of the `class` token into account nor knowing if what follows the `class` keyword is actually the class name.

Reply:
Hm, so what if we just collect all tokens from the current line, then change only the one T_STRING with the class name and concatenate them back into source code for this one line? Should be ugly but fool-proof, no?